test/server/processOperations.test.ts

```diff
@@ -461,4 +461,242 @@ describe('processOperations', () => {
+
+        const operation0 = buildOperation(
+            reducer,
+            document,
+            actions.setModelName({ name: '1' }),
+            0
+        );
+
+        const resultOp0 = await server.addOperation(
+            driveId,
+            documentId,
+            operation0
+        );
+
+        expect(resultOp0.status).toBe('SUCCESS');
+
+        document = (await server.getDocument(
+            driveId,
+            documentId
+        )) as DocumentModelDocument;
+
+        const operation1 = buildOperation(
+            reducer,
+            document,
+            actions.setModelName({ name: '2' }),
+            0
+        );
+
+        const resultOp1 = await server.addOperation(
+            driveId,
+            documentId,
+            operation1
+        );
+
+        expect(resultOp1.status).toBe('SUCCESS');
+
+        document = (await server.getDocument(
+            driveId,
+            documentId
+        )) as DocumentModelDocument;
+
+        const operation2 = buildOperation(
+            reducer,
+            document,
+            actions.setModelId({ id: '3' }),
+            0
+        );
+
+        const resultOp2 = await server.addOperation(
+            driveId,
+            documentId,
+            operation2
+        );
+
+        expect(resultOp2.status).toBe('SUCCESS');
+
+        document = (await server.getDocument(
+            driveId,
+            documentId
+        )) as DocumentModelDocument;
+
+        const operation3 = buildOperation(
+            reducer,
+            document,
+            actions.setModelId({ id: '4' }),
+            0
+        );
+
+        const resultOp3 = await server.addOperation(
+            driveId,
+            documentId,
+            operation3
+        );
+
+        expect(resultOp3.status).toBe('SUCCESS');
+
+        document = (await server.getDocument(
+            driveId,
+            documentId
+        )) as DocumentModelDocument;
+
+        const operations = document.operations.global.slice(-4);
+
+        expect(document.state.global).toMatchObject({
+            name: '2',
+            id: '4'
+        });
+        expect(operations).toMatchObject([
+            {
+                type: 'SET_MODEL_NAME',
+                input: { name: '1' },
+                index: 6,
+                skip: 6
+            },
+            {
+                type: 'SET_MODEL_NAME',
+                input: { name: '2' },
+                index: 7,
+                skip: 0
+            },
+            {
+                type: 'SET_MODEL_ID',
+                input: { id: '3' },
+                index: 8,
+                skip: 0
+            },
+            {
+                type: 'SET_MODEL_ID',
+                input: { id: '4' },
+                index: 9,
+                skip: 0
+            }
+        ]);
+    });
+
+    it('should resolve conlficts suign the right order for merge oprations (simulate clients conflict)', async () => {
```

Comment:
I can feel how tired you were when doing this ahah

typo: should resolve conflicts using the right order for merge operations

Reply:
added fix here: [8cb126d](https://github.com/powerhouse-inc/document-drive/pull/129/commits/8cb126d241bb8db0809b763990a0a45f2050c627)